tests/test_models/test_reservation_unit_pricing_updates.py

```diff
@@ -12,64 +15,263 @@
+
+    future_pricing_count = ReservationUnitPricing.objects.filter(status=PricingStatus.PRICING_STATUS_FUTURE).count()
+    assert future_pricing_count == 1  # No new pricings should be created, since a pricing after the change date exists
+
+
+def test_reservation_unit__update_pricings__tax_percentage__free_future_pricing_on_change_date():
+    pricing_1 = ReservationUnitPricingFactory.create(
+        begins=datetime.date(2024, 1, 1),
+        status=PricingStatus.PRICING_STATUS_ACTIVE,
+        tax_percentage__value=CURRENT_TAX,
+    )
+    ReservationUnitPricingFactory.create(
+        begins=TAX_CHANGE_DATE,
+        status=PricingStatus.PRICING_STATUS_FUTURE,
+        reservation_unit=pricing_1.reservation_unit,
+        tax_percentage__value=CURRENT_TAX,
+        pricing_type=PricingType.FREE,
+    )
+
+    update_reservation_unit_pricings_tax_percentage(str(TAX_CHANGE_DATE), str(CURRENT_TAX), str(FUTURE_TAX))
+
+    future_pricing_count = ReservationUnitPricing.objects.filter(status=PricingStatus.PRICING_STATUS_FUTURE).count()
+    assert future_pricing_count == 1  # No new pricings should be created, since a pricing on the change date exists
+
+
+def test_reservation_unit__update_pricings__tax_percentage__free_future_pricing_after_change_date():
+    pricing_1 = ReservationUnitPricingFactory.create(
+        begins=datetime.date(2024, 1, 1),
+        status=PricingStatus.PRICING_STATUS_ACTIVE,
+        tax_percentage__value=CURRENT_TAX,
+    )
+    ReservationUnitPricingFactory.create(
+        begins=datetime.date(2024, 9, 21),
+        status=PricingStatus.PRICING_STATUS_FUTURE,
+        reservation_unit=pricing_1.reservation_unit,
+        tax_percentage__value=CURRENT_TAX,
+        pricing_type=PricingType.FREE,
+    )
+
+    update_reservation_unit_pricings_tax_percentage(str(TAX_CHANGE_DATE), str(CURRENT_TAX), str(FUTURE_TAX))
+
+    future_pricing_count = ReservationUnitPricing.objects.filter(status=PricingStatus.PRICING_STATUS_FUTURE).count()
+    assert future_pricing_count == 2  # New pricing should be created, since the FREE pricing is after the change date
+
+    # A new pricing
+    future_pricing = ReservationUnitPricing.objects.filter(
+        status=PricingStatus.PRICING_STATUS_FUTURE,
+        begins=TAX_CHANGE_DATE,
+    ).first()
+    assert future_pricing.tax_percentage.value == FUTURE_TAX
+    assert future_pricing.highest_price == pricing_1.highest_price
+    assert future_pricing.highest_price_net < pricing_1.highest_price_net
+
+
+def test_reservation_unit__update_pricings__tax_percentage__different_tax_percentage_is_ignored():
+    reservation_unit = ReservationUnitFactory.create()
+    ReservationUnitPricingFactory.create(
+        begins=datetime.date(2024, 1, 1),
+        status=PricingStatus.PRICING_STATUS_ACTIVE,
+        reservation_unit=reservation_unit,
+        tax_percentage__value=CURRENT_TAX,
```

Comment:
Tax percentage should be different, this tests the pricing type `FREE` case (copy pasta from below).